Helper/AlgoliaHelper.php

```diff
@@ -241,6 +241,11 @@ public function setSettings(
             $settings = $this->mergeSettings($indexName, $settings, $mergeSettingsFrom);
         }
 
+        // Removing synonyms from the settings to avoid duplication (synonyms must be managed within the algolia dashboard)
+        if (isset($settings['synonyms'])) {
+            unset($settings['synonyms']);
```

Comment:
We already have an operation that removes problematic attributes from settings at `\Algolia\AlgoliaSearch\Helper\AlgoliaHelper::mergeSettings`. I'm thinking we should centralize the logic there and perhaps make it a discrete function. WDYT?

Reply:
Makes sense ! Updating it 👍